sample/SipInterconnect/README.md

```diff
@@ -0,0 +1,238 @@
+
The SIP Gateway is comprised of a session border controller (SBC), a Signaling Gateway, and a Media Mixer. The SBC component handles security as well as the protocol and codec adaptation. The Signaling Gateway converts the SIP primitives into the corresponding OpenTok primitives. The Media Mixer mixes the audio flows from multiple participants in OpenTok session in order to forward them as a single composite audio flow.
+
+_**NOTE**: If media inactivity lasts for more than 5 minutes, the call will be automatically closed unless session timers are used at the SIP level. As a security measure, any SIP call longer than 6 hours will be automatically closed by OpenTok._
+
+
+### Security considerations
+
+There are some best practices recommended by TokBox when using this new SIP Interface with
+your SIP Servers. They try to mitigate the possible attacks by providing the mechanisms to
+authenticate and authorize that the SIP calls received in your server are legitimate and to
+encrypt all the signaling and media:
+
+* Use TLS and enable secure calls (SRTP) for signaling to avoid the possibility of
+  intercepting the communications.
+
+* Enable SIP authentication on your server. Otherwise, anyone who knows your SIP URI could
+  send calls to your server.
+
+If required, you can also block the traffic not coming from the IP addresses of the OpenTok
+SIP gateway, which uses the following IP blocks:
+
+* 52.200.60.16 &ndash; 52.200.60.31
+* 52.41.63.240 &ndash; 52.41.63.255
+* 52.51.63.16 &ndash; 52.51.63.31
+
+Be aware that TokBox can add new IP blocks in the future, and you
+will need to change your configuration.
+
+### Technical details
+
+**RFC3550 (RTP/RTCP) support:** Media traffic can be encrypted (SRTP) or non-­encrypted
+(plain RTP). In case of encryption, both DTLS and SDES protocols are supported.
+
+**Codec support:** The OpenTok SIP gateway supports the OPUS, G.711, and G.722 audio codecs.
+
+**Signaling:** The OpenTok SIP gateway supports RFC3561 (SIP) over UDP, TCP, and TLS.
+Contact TokBox if you need information or support for any specific extension.
+
+The OpenTok SIP gateway will not accept any SIP message coming from the a third-party SIP
+platform unless it is part of a SIP dialog initiated by the OpenTok SIP gateway.
+Calls initiated with the OpenTok SIP gateway can be put on ­hold using either a `re-­INVITE`
+with the `sendonly/inactive` direction in the SDP or a `re-­INVITE` with port 0 in the SDP.
+
+**Other considerations:** Early media is disabled, and DTMFs are not currently supported.
+
+
+### Additional Information
+
+For additional information see [SIP Interconnect](https://www.tokbox.com/developer/guides/sip/) at the TokBox Developer Center. 
+
+
+
+## Prerequisites
+
+To be prepared to develop your OpenTok SIP Interconnect app:
+
+1. Review the [OpenTok.js](https://tokbox.com/developer/sdks/js/) requirements.
```

Comment:
i feel like the information about running the app should be as close to the top as possible.